### PR TITLE
Neptune

### PR DESCRIPTION
**Pull request by whyshreddy**
# Description
Added neptune logging into train.py 
User story # (US link)
https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/149

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X ] Test A -tested locally if neptune logging works

# Checklist:

- [ x] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests
- [ ] New and existing unit tests pass locally with my changes
- [ ] My Code is review by 1 People

### CODE REVIEW

File: src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/src/train.py [hunk 1]
```diff
@@ -217,6 +257,8 @@ def on_epoch_end(self, epoch, logs=None):
     callbacks=training_callbacks,
     verbose=2
 )
+if os.path.exists(best_model_path):
```
Comment (markus-hinsche) [inline src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/src/train.py:260]: Why do we only log the best model in neptune after the training is finished?
Can we use a Callback to do it whenever there is a new model?

File: src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/src/train.py [hunk 2]
```diff
@@ -155,16 +164,25 @@ def tf_flip(image):
 
 # Note: Now the datasets are prepared.
 
+#parameters info to log into neptune
+training_params = CONFIG
```
Comment (markus-hinsche) [inline src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/src/train.py:168]: Can we  put this line closer to it's usage for more readability?

File: src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/src/train.py [hunk 3]
```diff
@@ -174,6 +192,26 @@ def on_epoch_end(self, epoch, logs=None):
...
+neptune.create_experiment(
+    params={**model_params, **training_params},
+    # dataset type, dataset version,quarter,user story
+    tags=['depthmap', 'anon-mini-95k', '2020Q4', 'US149'],
+    description=description
+)
+#logging the model to the neptune
+model.summary(print_fn=lambda x: neptune.log_text('model_summary', x))
+
+# Pushes metrics and losses into the run on Neptune
```
Comment (markus-hinsche) [inline src/models/CNNDepthMap/CNNDepthMap-height/q3-depthmap-plaincnn-height/src/train.py:205]: I don't understand what this comment belongs to. I suggest to not add newlines after a comment for better readability.

**Review by markus-hinsche — CHANGES_REQUESTED** [review]
Thanks for submitting the PR. 

One problem that I see is that this PR changes 44 lines in `train.py`. This makes `train.py` more complicated.
I suggest to refactor such that I we only have to add 1 or 2 lines to an experiment to include neptune: We could put a util function into utils.py and call it in train.py s.t. train.py only has to add 1-2lines

**Comment by AI-Guru** [comment]
I would suggest putting the NeptuneLogger into a standalone module for reuse. Also the whole neptune.create_experiment(...) business could go into the constructor of the NeptuneLogger. I think this is what Markus also would like.

**Comment by markus-hinsche** [comment]
Temporarily closing due to inactivity

**Pull request closed without merging**